airflow/providers/microsoft/azure/provider.yaml

```
@@ -56,8 +56,7 @@ dependencies:
   - azure-mgmt-datafactory>=1.0.0,<2.0
   - azure-mgmt-datalake-store>=0.5.0
   - azure-mgmt-resource>=2.2.0
-  # limited due to https://github.com/Azure/azure-sdk-for-python/pull/18801  implementation released in 12.9
-  - azure-storage-blob>=12.7.0,<12.9.0
+  - azure-storage-blob>=12.9.0,<12.14.0
```

I think the workaround should continue to work after 12.14.0 (unless they change things _drastically_, which is very unlikely)

```suggestion
  - azure-storage-blob>=12.9.0
```

---

sure we can do that.. I planned to bump to 12.14.0 to avoid this extra code. Are you OK with leaving it as is?

---

https://github.com/apache/airflow/pull/25426/commits/e9d7714d0205bfc0686354f08ed0df057828f516